proxy/pkg/zdmproxy/clusterconn.go

```diff
@@ -252,6 +263,25 @@ func (cc *ClusterConnector) runResponseListeningLoop() {
 				}
 			}
 
+			// when there's a protocol error, we cannot rely on the returned stream id, the only exception is
+			// when it's a UnsupportedVersion error, which means the Frame was properly parsed but the proxy doesn't
+			// support the protocol version and in that case we can proceed with releasing the stream id in the mapper
+			if response != nil && response.Header.StreamId >= 0 && (err != nil || errCode == ProtocolErrorUnsupportedVersion) {
```

Comment:
I'm trying to figure out why the `err != nil` here, can you clarify?

Reply:
It was supposed to be `err == nil` as we sync'ed offline (I understand you're changing that yourself with another commit you're doing here. Thanks).